app/downloadshandler.py

```diff
@@ -59,7 +59,7 @@ async def get(self, request):
 
     user_role = await saml.get_role_id(request)
 
-    if download_permission(user_role):
+    if has_download_permission(user_role):
```

Comment:
It doesn't need to be written this way around - you should use `if not has_download_permission(user_role)` so you don't need to use this odd dangling-true-branch.

Reply:
Have adjusted